engine/src/test/java/org/entando/entando/aps/system/services/component/ComponentServiceTest.java

```diff
@@ -0,0 +1,140 @@
+        services.add(this.otherMockService);
+        this.componentService = new ComponentService(this.services);
+    }
+    
+    @Test
+    void extractComponentUsageDetailsWithErrorOnDto() throws EntException {
+        List<Map<String, String>> request = List.of(
+                Map.of("type", "myType", "code", "service"));
+        Mockito.when(mockService.getObjectType()).thenReturn("myType");
+        Mockito.when(mockService.getComponentDto(Mockito.anyString())).thenThrow(new EntException("error"));
+        Assertions.assertThrows(RestServerError.class, () -> {
+            this.componentService.extractComponentUsageDetails(request);
+        });
+    }
+    
+    @Test
+    void extractComponentUsageDetailsWithError() throws EntException {
+        List<Map<String, String>> request = List.of(
+                Map.of("type", "customType", "code", "service"));
+        Mockito.when(mockService.getObjectType()).thenReturn("customType");
+        Mockito.when(mockService.getComponentDto(Mockito.anyString())).thenReturn(Optional.of(Mockito.mock(IComponentDto.class)));
+        Mockito.when(mockService.getComponentUsageDetails(Mockito.anyString(), Mockito.any())).thenThrow(new EntRuntimeException("error"));
+        Assertions.assertThrows(RestServerError.class, () -> {
+            this.componentService.extractComponentUsageDetails(request);
+        });
+    }
+    
+    @Test
+    void deleteComponentWithError() throws EntException {
+        List<Map<String, String>> request = List.of(
+                Map.of("type", "type", "code", "service"));
+        Mockito.when(mockService.getObjectType()).thenReturn("type");
+        Mockito.when(mockService.getComponentDto(Mockito.anyString())).thenThrow(new EntException("error"));
+        Assertions.assertThrows(RestServerError.class, () -> {
+            this.componentService.deleteInternalComponents(request);
+        });
+    }
+    
+    @Test
+    void deleteNonExistingComponent() throws EntException {
+        List<Map<String, String>> request = List.of(
+                Map.of("type", "type", "code", "service"),
+                Map.of("type", "otherType", "code", "internalReference"));
+        Mockito.when(mockService.getObjectType()).thenReturn("type");
+        ComponentUsageEntity reference = new ComponentUsageEntity("otherType", "internalReference");
+        PagedMetadata<ComponentUsageEntity> pm = new PagedMetadata<>(new RestListRequest(), 1);
+        pm.setBody(List.of(reference));
+        Mockito.when(mockService.getComponentDto(Mockito.anyString())).thenReturn(Optional.empty());
+        ComponentDeleteResponse response = this.componentService.deleteInternalComponents(request);
+        Assertions.assertEquals(ComponentDeleteResponse.STATUS_PARTIAL_SUCCESS, response.getStatus());
+        Assertions.assertEquals(1, response.getComponents().size());
+        Assertions.assertEquals(ComponentDeleteResponse.STATUS_FAILURE, response.getComponents().get(0).get("status"));
```

Comment:
please don't use a constant from the src code, in the test you should use a test constant (or a string) because the tests are specifications, so if you change the src code constant with a refactor by error the tests will warn you

Reply:
@mirkoPan please fix also this

Reply:
done